File: flutter_app/lib/features/feed/domain/entities/post.dart
import 'package:flutter/foundation.dart';

@immutable
class PostAttachment {
  const PostAttachment({
    required this.id,
    required this.filename,
    required this.mimeType,
    required this.sizeBytes,
    required this.url,
  });

  final String id;
  final String filename;
  final String mimeType;
  final int sizeBytes;
  final String url;
}

@immutable
class Post {
  const Post({
    required this.id,
    required this.groupId,
    required this.authorId,
    required this.authorName,
    required this.body,
    required this.pinned,
    required this.createdAt,
    required this.updatedAt,
    this.attachments = const [],
  });

  final String id;
  final String groupId;
  final String authorId;
  final String authorName;
  final String body;
  final bool pinned;
  final List<PostAttachment> attachments;
  final DateTime createdAt;
  final DateTime updatedAt;

  Post copyWith({
    String? body,
    bool? pinned,
    List<PostAttachment>? attachments,
  }) =>
      Post(
        id: id,
        groupId: groupId,
        authorId: authorId,
        authorName: authorName,
        body: body ?? this.body,
        pinned: pinned ?? this.pinned,
        attachments: attachments ?? this.attachments,
        createdAt: createdAt,
        updatedAt: updatedAt,
      );
}

File: backend/internal/domain/post.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Post struct {
	ID        uuid.UUID  `db:"id"`
	GroupID   uuid.UUID  `db:"group_id"`
	AuthorID  uuid.UUID  `db:"author_id"`
	Body      string     `db:"body"`
	Pinned    bool       `db:"pinned"`
	CreatedAt time.Time  `db:"created_at"`
	UpdatedAt time.Time  `db:"updated_at"`
	DeletedAt *time.Time `db:"deleted_at"`
}

type PostAttachment struct {
	ID        uuid.UUID `db:"id"`
	PostID    uuid.UUID `db:"post_id"`
	ObjectKey string    `db:"object_key"`
	Filename  string    `db:"filename"`
	MimeType  string    `db:"mime_type"`
	SizeBytes int64     `db:"size_bytes"`
	CreatedAt time.Time `db:"created_at"`
}

// PostWithMeta — пост плюс имя автора и вложения (для ответа API).
type PostWithMeta struct {
	Post
	AuthorName string            `db:"author_name"`
	Attachments []PostAttachment `db:"-"`
}
